model/project_parser.go

```diff
@@ -82,6 +82,7 @@ type ParserProject struct {
 	EarlyTermination   *YAMLCommandSet            `yaml:"early_termination,omitempty" bson:"early_termination,omitempty"`
 	CallbackTimeout    *int                       `yaml:"callback_timeout_secs,omitempty" bson:"callback_timeout_secs,omitempty"`
 	Modules            []Module                   `yaml:"modules,omitempty" bson:"modules,omitempty"`
+	AutoUpdateModules  *bool                      `yaml:"auto_update_modules,omitempty" bson:"auto_update_modules,omitempty"`
```

Comment:
This is slightly complicated because really this does seem like it should only be relevant to PR patches 😅 otherwise to test a module at a specific revision you'd need to go and remove this from the yaml to do that (although maybe that's fine since there's a workaround). An alternative idea could be to have this be part of the PR testing section on the project page, configurable that way? (It would require a spruce PR but they could probably do it pretty fast, especially since this impacts spruce in particular) That could make it very clear that this just impacts PR testing.

Reply:
I actually think it'd be useful to have this configurable for `evergreen patch` as well (@khelif96 mentions it in the JIRA comments) - if a module branch has commits ahead of even the most recent project commit then the currently the only way to run a patch with the updated module if the project is behind is to add an empty commit to the project.

Maybe we can allow CLI and PR patches to set this value individually? If we're looking to over time move configurations from the UI to YAML it might be preferable to put new configurations moving forward on the YAML side since this might be something a non-admin developer might want to toggle back and forth. Maybe we can move these flags into the `modules` YAML structure for more simplicity?

Reply:
Hmmm okay I see. I think maybe making this configurable at the module yaml level does make sense then.

Reply:
(Adding a comment with the review button so it shows me as having responded)

Reply:
(Adding a comment with the review button so it shows me as having responded)

Reply:
Sounds good, per our offline talk I've moved `auto_update` to the module level, and now for patch finalization we fetch the base revision manifest, update it for the modules with `auto_update` , and then re-insert the manifest as opposed to constructing an entirely new manifest for a patch.